src/careamics_restoration/engine.py

```diff
@@ -524,3 +530,77 @@ def __del__(self) -> None:
                 if isinstance(handler, FileHandler):
                     self.logger.removeHandler(handler)
                     handler.close()
+
+    def save_as_bioimage(
+        self, output_zip: Union[Path, str], model_specs: Optional[dict] = None
+    ) -> BioimageModel:
+        """Export the current model to BioImage.io model zoo format.
+
+        Parameters
+        ----------
+        output_zip (Union[Path, str]): Where to save the model zip file.
+        model_specs (Optional[dict]): a dictionary that keys are the bioimage-core
+        `build_model` parameters.
+        If None then it will be populated up by the model default specs.
+        """
+        workdir = self.cfg.working_directory
+        # load the best check point
+        checkpoint_path = workdir.joinpath(
+            f"{self.cfg.experiment_name}_best.pth"
+        ).absolute()
+        checkpoint = torch.load(checkpoint_path)
+        # save the model state_dict separately.
+        weight_path = workdir.joinpath("model_weights.pth")
+        torch.save(checkpoint["model_state_dict"], weight_path)
+        # get in/out samples' files
+        test_inputs, test_outputs = self._get_sample_io_files()
+
+        specs = get_default_model_specs(self.cfg.algorithm.loss)
+        if model_specs is not None:
+            specs.update(model_specs)
+
+        specs.update(
```

Comment:
We should add a tag `2D` or `3D`, depending on the axes.

Reply:
Also axes should be set by the actual `Engine`, not be default.

Reply:
I see, ok.

Reply:
In the careamics config, the axes could be `STCZYX`, and I assume that we don't have the "batch" axes there, right?  
Because data for a conv net sould be at least 4D, and if I replace the "S" for "b" we get "byx" which is just 3D.

Reply:
@CatEek 

The channel dimension should be there as well (bcyx). That might be a problem because we do "pre-processing" on the axes and add the S and C dimensions if necessary. But the bioimage.io core might not.

Although I am pretty sure this was also the case in the napari plugin. So maybe that is all good. Maybe look at how things are done in the napari plugin because it should be quite similar.